ui/app/scripts/controllers/newDeploymentCtrl.js

```diff
@@ -445,26 +614,113 @@ angular.module('apollo')
+		     apolloApiService.getAllEnvironmentsStacks().then(function(response) {
+             var tempEnvironmentsStack = {};
+		     response.data.forEach(function(environmentsStack) {
+                tempEnvironmentsStack[environmentsStack.id] = environmentsStack;
+                addStackToDisplayEnvironments(environmentsStack);
+
+                var environmentsNames = [];
+                environmentsStack.environments.forEach(function(environmentId) {
+                    var environment = allEnvironments.find(x => x.id === environmentId);
+                    var environmentName = environment.name;
+                    environmentsNames = environmentsNames.concat(environmentName);
+                });
 
-            if (previousEnvironmentIds !== null) {
-                $scope.selectedEnvironments = $scope.possibleEnvironments.filter(function(a){return previousEnvironmentIds.indexOf(a.id) > -1})
+                $scope.environmentsInStack[environmentsStack.id] = environmentsNames;
+             });
+             $scope.possibleEnvironmentsStacks = tempEnvironmentsStack;
+             updateFinalDisplayEnvironmentsAndStacks();
+        });
+        })
+
+        function parseAllEnvironments(response) {
+            var tempEnvironment = {};
+            response.data.forEach(function(environment) {
+                tempEnvironment[environment.id] = environment;
+                addEnvironmentToDisplayEnvironments(environment);
+            });
+            $scope.possibleEnvironments = tempEnvironment;
+            allEnvironments = response.data;
+            updateFinalDisplayEnvironmentsAndStacks();
+
+           return Promise.resolve();
+        }
+
+		function addStackToDisplayEnvironments(stack) {
+            stack.isStack = true;
+            stack.orderId = -1;
+            environmentsAndStackForDisplayWorking = environmentsAndStackForDisplayWorking.concat(stack);
+        }
+
+        function addEnvironmentToDisplayEnvironments(environment) {
+            environment.isStack = false;
+            environment.orderId = 1;
+            environmentsAndStackForDisplayWorking = environmentsAndStackForDisplayWorking.concat(environment);
+        }
+
+         function updateFinalDisplayEnvironmentsAndStacks() {
+            if ($scope.possibleEnvironments !== null && $scope.possibleEnvironmentsStacks !== null) {
+                    $scope.environmentsAndStackForDisplay = environmentsAndStackForDisplayWorking;
             }
-		});
+         }
+
+		apolloApiService.getAllServices()
+		    .then(parseAllServices)
+		    .then(() => {
+		        apolloApiService.getAllServicesStacks().then(function(response) {
+                    var tempServicesStack = {};
+                    response.data.forEach(function(servicesStack) {
+                       tempServicesStack[servicesStack.id] = servicesStack;
+                       addStackToDisplayServices(servicesStack);
+
+                       var servicesNames = [];
```

Comment:
Same comment here as the environment

Reply:
what?

Reply:
It's the exact same code.. all comments on the previous environment section applies here